kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java

```diff
@@ -530,23 +529,23 @@ private void removeDeletedFromWireGraphProperty(final String pid) throws GwtKura
         final ConfigurationAdmin configAdmin = ServiceLocator.getInstance().getService(ConfigurationAdmin.class);
         final WireHelperService wireHelperService = ServiceLocator.getInstance().getService(WireHelperService.class);
         try {
+            final String filter = "(" + KURA_SERVICE_PID + "=" + pid + ")";
+            ServiceUtil.waitForService(filter, TIMEOUT, SECONDS);
             final String servicePid = wireHelperService.getServicePid(pid);
-            Configuration conf = null;
-            if (servicePid != null) {
-                conf = configAdmin.getConfiguration(servicePid);
+            if (servicePid == null) {
+                return;
             }
-            Dictionary<String, Object> props = null;
-            if (conf != null) {
-                props = conf.getProperties();
+            Configuration conf = configAdmin.getConfiguration(servicePid);
+            if (conf == null) {
+                return;
             }
+            Dictionary<String, Object> props = conf.getProperties();
             if (props != null) {
                 props.remove(DELETED_WIRE_COMPONENT);
-            }
-            if (conf != null) {
                 conf.update(props);
             }
-        } catch (final IOException e) {
-            // no need
+        } catch (final Exception exception) {
+            throw new GwtKuraException(GwtKuraErrorCode.INTERNAL_ERROR, exception);
```

Comment:
You can catch three exceptions in two catch blocks - `IOException, InvalidSyntaxException` and `InterruptedException`

`IOException, InvalidSyntaxException`  -> `CONFIGURATION_ERROR` (as it is related to ConfigAdmin and filter)
`InterruptedException` -> `PROCESS_EXECUTION_ERROR`